### DISCUSSION TITLE
HxInputFileDropZone Validation and Display files on Page load?

### DISCUSSION THREAD

**Discussion by bluisanajmr** [havit/Havit.Blazor#845] [Q&A]
Hello, 

I am using the HxInputFileDropZone to allow users to upload a logo for their company.  The HxInputFileZone is on my CompanyEdit form which also contains all other inputs for a company (name, address, etc..).  This page uses an EditContext and EditForm for inserting and editing data with validation messages for all inputs with validation messages. 

Is it possible to have the HxInputFileZone display the name of a saved uploaded company logo when a users navigates to the companyedit page for their company with existing data?  Is it possible to display a required validation message for the HxInputFileZone if no files are selected and the form is submitted?

**Comment by hakenr** [comment]
@bluisanajmr
1. You can use `NoFilesTemplate` to display any content of your choice when no file is selected.
2. Use `HxValidationMessage` and bind it to a property of your model that indicates whether a file is selected.
